src/App.vue

```diff
@@ -0,0 +1,236 @@
+  name: "App",
+  components: {
+    Card,
+  },
+  data() {
+    return {
+      fullScreen: false,
+    };
+  },
+  setup() {
+    const cardList = ref([]);
+    const userSelection = ref([]);
+
+    const MatchStatus = computed(() => {
+      if (remainingPairs.value === 0) {
+        return "Gracz Wygrywa";
+      } else {
+        return `Pozostałe Pary: ${remainingPairs.value}`;
+      }
+    });
+
+    const remainingPairs = computed(() => {
+      const remainingCards = cardList.value.filter(
+        (card) => card.matched === false
+      ).length;
+      return remainingCards / 2;
+    });
+
+    const shuffleCards = () => {
+      cardList.value = _.shuffle(cardList.value);
+    };
+
+    const restartGame = () => {
+      shuffleCards();
+
+      cardList.value = cardList.value.map((card, index) => {
+        return {
+          ...card,
+          position: index,
+          matched: false,
+          visible: false,
+        };
+      });
+    };
+    const openFullScreen = () => {
+      if (document.fullscreenElement) {
+        document
+          .exitFullscreen()
+          .then(() => console.log("Document Exited from Full screen mode"))
+          .catch((err) => console.error(err));
+      } else {
+        document.documentElement.requestFullscreen();
+      }
+    };
+
+    const cardItems = [
+      "#66FFFF",
+      "#990033",
+      "#FFFF00",
+      "#0E0E0E",
+      "#76FF03",
+      "#1029A1",
+      "#696969",
+      "#CF7E3C",
+    ];
+    cardItems.forEach((item) => {
+      cardList.value.push({
+        value: item,
+        visible: false,
+        matched: false,
+      });
+
+      cardList.value.push({
+        value: item,
+        visible: false,
+        matched: false,
+      });
+    });
+    cardList.value = cardList.value.map((card, index) => {
+      return {
+        ...card,
+        position: index,
+      };
+    });
+
+    const flipCard = (payload) => {
+      cardList.value[payload.position].visible = true;
+      if (userSelection.value[0]) {
+        if (userSelection.value[0].position === payload.position) {
+          return;
+        } else {
+          userSelection.value[1] = payload;
+        }
+      } else {
+        userSelection.value[0] = payload;
+      }
+    };
+
+    watch(
+      userSelection,
+      (currentValue) => {
+        if (currentValue.length === 2) {
+          const cardFirst = currentValue[0];
+          const cardSecond = currentValue[1];
+          if (cardFirst.faceValue === cardSecond.faceValue) {
+            setTimeout(() => {
+              cardList.value[cardFirst.position].matched = true;
+              cardList.value[cardSecond.position].matched = true;
+            }, 500);
+          } else {
+            setTimeout(() => {
+              cardList.value[cardFirst.position].visible = false;
+              cardList.value[cardSecond.position].visible = false;
+            }, 500);
+          }
+
+          userSelection.value.length = 0;
```

Comment:
w data powinno być prop `selectedCardAmount` i tutaj robisz `this.selectedCardAmount = 0`